src/main/scala/li/cil/oc/integration/appeng/ConverterPattern.java

```diff
@@ -1,38 +1,62 @@
-                for (int i = 0; i < p.getInputs().length; ++i) {
+                final NBTTagCompound encodedValue = is.getTagCompound();
+                final NBTTagList inTag = encodedValue.getTagList("in", 10);
+                final NBTTagList outTag = encodedValue.getTagList("out", 10);
+
+                Map[] inputs = new Map[inTag.tagCount()];
+                for (int i = 0; i < inTag.tagCount(); i++) {
                     inputs[i] = new HashMap<>();
-                    if (p.getInputs()[i] == null) continue;
-                    ItemStack input = p.getInputs()[i].getItemStack();
-                    inputs[i].put("name", input.getItem().getItemStackDisplayName(input));
-                    inputs[i].put("count", input.stackSize);
+                    final NBTTagCompound tag = inTag.getCompoundTagAt(i);
+                    final ItemStack inputItem = Platform.loadItemStackFromNBT(tag);
+                    if (inputItem != null) {
+                        inputs[i].put("name", inputItem.getItem().getItemStackDisplayName(inputItem));
+                        if (tag.getLong("Cnt") > 0) {
+                            inputs[i].put("count", tag.getLong("Cnt"));
+                        } else {
+                            inputs[i].put("count", inputItem.stackSize);
+                        }
+                    }
                 }
-                output.put("inputs", inputs);
-                Map[] results = new Map[p.getOutputs().length];
-                for (int i = 0; i < p.getOutputs().length; ++i) {
+
+                Map[] results = new Map[outTag.tagCount()];
+                for (int i = 0; i < outTag.tagCount(); i++) {
                     results[i] = new HashMap<>();
-                    if (p.getOutputs()[i] == null) continue;
-                    ItemStack result = p.getOutputs()[i].getItemStack();
-                    results[i].put("name", result.getItem().getItemStackDisplayName(result));
-                    results[i].put("count", result.stackSize);
+                    final NBTTagCompound tag = outTag.getCompoundTagAt(i);
+                    final ItemStack outputItem = Platform.loadItemStackFromNBT(tag);
+                    if (outputItem != null) {
+                        results[i].put("name", outputItem.getItem().getItemStackDisplayName(outputItem));
+                        if (tag.getLong("Cnt") > 0) {
+                            results[i].put("count", tag.getLong("Cnt"));
+                        } else {
+                            results[i].put("count", outputItem.stackSize);
+                        }
+                    }
                 }
-                output.put("outputs", results);
-                output.put("isCraftable", p.isCraftable());
+                    output.put("inputs", inputs);
+                    output.put("outputs", results);
+
+                    PatternHelper p = new PatternHelper(is, null);
+                    output.put("isCraftable", p.isCraftable());
```

Comment:
If you take everything directly from NBT why leave PatternHelper here

Reply:
Good point, I assumed the PatternHelper did something fancy to figure out if a pattern is craftable or not but looks like it is just a boolean in the NBT.